lib/ansible/galaxy/api.py

```diff
@@ -904,3 +904,33 @@ def get_collection_signatures(self, namespace, name, version):
             return []
         else:
             return [signature_info["signature"] for signature_info in signatures]
+
+    @g_connect(['v2', 'v3'])
+    def search_collections(self, term, **kwargs):
+        """
+        Searches the galaxy server for matching collections or namespaces
+
+        :param term :  term to search
+        :param name: The collection name.
+        :param version: Version of the collection to get the information for.
+        :return: A list of signature strings.
+        """
+
+        author = kwargs.get('author', None)
+        namespace = kwargs.get('namespace', None)
+        platforms = kwargs.get('platforms', None)
+        page_size = kwargs.get('page_size', None)
+        tags = kwargs.get('tags', None)
+
+        # api/internal/ui/search/?keywords=docker&order_by=-relevance&page=1&deprecated=false&type=collection
+        # https://galaxy.ansible.com/search?keywords=<term>&order_by=-relevance&deprecated=false&type=collection&page=1
+        # api_path = self.available_api_versions.get('v3', self.available_api_versions.get('v2'))
+        # url_paths = [self.api_server, api_path, 'collections',  '/?search=%s' % term]
+        url_paths = ['https://galaxy.ansible.com', 'api', 'internal', 'ui', 'search', '/?type=collection&deprecated=false&keywords=%s' % term]
```

Comment:
Could use self.api_server instead:
```suggestion
        url_paths = [self.api_server, 'internal', 'ui', 'search', '/?type=collection&deprecated=false&keywords=%s' % term]
```

Unless it's hardcoded because it doesn't work with v3?

Reply:
nor v2

Reply:
isn't v2 community galaxy collections? v1 is roles

But noticed the conversation about internal/ui not being a supported, so it doesn't really matter.

Reply:
for roles we still use the v1, i tried equivalent v2 and v3 but could not find, will resolve with help of galaxy team